snoop/data/analyzers/archives.py

```diff
@@ -195,21 +193,28 @@ def unarchive(blob):
 
 def archive_walk(path):
     """Generates simple dicts with archive listing for the archive. """
-
-    for thing in path.iterdir():
-        if thing.is_dir():
-            yield {
-                'type': 'directory',
-                'name': thing.name,
-                'children': sorted(
-                    list(archive_walk(thing)),
-                    key=lambda c: c['name'],
-                ),
+    res = []
+    for root, dirs, files in os.walk(path, topdown=True):
+        children = []
+        for d in dirs:
+            dir_info = {
+                'type' : 'directory',
+                'name' :  os.path.relpath(d),
```

Comment:
this should be only the last path segment (in your case `d.name` - look at the code you deleted) and not some path:

```                
                'type': 'directory',
                'name': thing.name,
```